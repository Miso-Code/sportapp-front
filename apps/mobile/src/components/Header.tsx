import React, { ComponentProps } from 'react'
import { View, StyleSheet } from 'react-native'
import { Stack } from 'expo-router/stack'
import { router } from 'expo-router'

import { IconButton, Text } from 'react-native-paper'
import { SafeAreaProvider } from 'react-native-safe-area-context'

const Header = ({
	back,
	options,
	route
}: ComponentProps<
	ComponentProps<typeof Stack.Screen>['options']['header']
>) => {
	const title =
		(route?.params as { title: string } | undefined)?.title || options.title
	return (
		<SafeAreaProvider style={styles.safeArea}>
			<View style={styles.header} testID='header'>
				{back && (
					<IconButton
						testID='back'
						icon='chevron-left'
						size={35}
						onPress={() => router.back()}
					/>
				)}
				<Text
					testID='title'
					variant='displaySmall'
					style={styles.headerTitle}
					numberOfLines={1}
					ellipsizeMode='tail'>
					{title}
				</Text>
			</View>
		</SafeAreaProvider>
	)
}

const styles = StyleSheet.create({
	header: {
		flex: 1,
		flexDirection: 'row',
		justifyContent: 'flex-start',
		alignItems: 'center',
		marginTop: 45,
		gap: 5,
		backgroundColor: '#f2f2f2'
	},
	headerTitle: {
		flex: 1,
		fontSize: 34,
		fontWeight: 'normal',
		height: 50,
		maxWidth: '70%'
	},
	safeArea: {
		backgroundColor: '#f2f2f2'
	}
})

export default Header
