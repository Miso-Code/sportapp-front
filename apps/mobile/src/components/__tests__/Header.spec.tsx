import React from 'react'
import renderer, { ReactTestRenderer, act } from 'react-test-renderer'
import { router } from 'expo-router'

import Header from '../Header'

jest.mock('expo-router')
jest.mock('react-native-safe-area-context')

describe('Header', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(
			<Header
				options={{}}
				route={{ key: '', name: '' }}
				navigation={undefined}
			/>
		)
	})

	afterEach(() => {
		jest.clearAllMocks()
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render a back button', () => {
		act(() => {
			component.update(
				<Header
					options={{ title: 'Title' }}
					route={{ key: '', name: '' }}
					navigation={undefined}
					back={{
						title: 'test'
					}}
				/>
			)
		})
		expect(component.root.findByProps({ testID: 'back' })).toBeTruthy()
	})

	it('should call router.back when back button is pressed', () => {
		act(() => {
			component.update(
				<Header
					options={{ title: 'Title' }}
					route={{ key: '', name: '' }}
					navigation={undefined}
					back={{
						title: 'test'
					}}
				/>
			)
		})
		component.root.findByProps({ testID: 'back' }).props.onPress()
		expect(router.back).toHaveBeenCalled()
	})
})
