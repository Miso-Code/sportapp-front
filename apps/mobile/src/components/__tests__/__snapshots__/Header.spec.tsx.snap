// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "flex": 1,
        "flexDirection": "row",
        "gap": 5,
        "justifyContent": "flex-start",
        "marginTop": 45,
      }
    }
    testID="header"
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 36,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 44,
            },
            {
              "flex": 1,
              "fontSize": 34,
              "fontWeight": "normal",
              "height": 50,
              "maxWidth": "70%",
            },
          ],
        ]
      }
      testID="title"
    />
  </View>
</View>
`;
